File: Lab01/TokenType.cs
namespace Lab01;

public enum TokenType
{
    NUMBER,
    ID,
    STRING,
    VAR,

    PRINT,
    IF,
    ELSE,
    WHILE,

    // Operators
    PLUS,
    MINUS,
    STAR,
    SLASH, // + - * /
    EQ,
    EQEQ,
    EXCL,
    NEQ, // = == ! !=
    LT,
    GT,
    LTEQ,
    GTEQ, // < > <= >=
    AND,
    OR, // && ||

    // Grouping & Punctuation
    LPAREN,
    RPAREN, // ( )
    LBRACE,
    RBRACE, // { }
    SEMICOLON, // ;

    EOF // end of input
}

File: src/mini_compiler/token_type.py
from enum import Enum, auto


class TokenType(Enum):
    NUMBER = auto()
    ID = auto()
    STRING = auto()
    VAR = auto()

    PRINT = auto()
    IF = auto()
    ELSE = auto()
    WHILE = auto()

    PLUS = auto()
    MINUS = auto()
    STAR = auto()
    SLASH = auto()
    EQ = auto()
    EQEQ = auto()
    EXCL = auto()
    NEQ = auto()
    LT = auto()
    GT = auto()
    LTEQ = auto()
    GTEQ = auto()
    AND = auto()
    OR = auto()

    LPAREN = auto()
    RPAREN = auto()
    LBRACE = auto()
    RBRACE = auto()
    SEMICOLON = auto()

    EOF = auto()
